analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -128,14 +130,127 @@ public static SyntaxNode WalkUpParentheses(this SyntaxNode node)
             return node;
         }
 
-        private static string GetUnknownType(SyntaxKind kind)
+        /// <summary>
+        /// Finds the syntactic complementing <see cref="SyntaxNode"/> of an assignment with tuples.
+        /// <code>
+        /// var (a, b) = (1, 2);      // if node is a, 1 is returned and visa versa.
+        /// (var a, var b) = (1, 2);  // if node is 2, var b is returned and visa versa.
+        /// </code>
+        /// <paramref name="node"/> must be an <see cref="ArgumentSyntax"/> of a tuple or some variable designation of a <see cref="SyntaxKindEx.DeclarationExpression"/>.
+        /// </summary>
+        public static SyntaxNode FindAssignmentComplement(this SyntaxNode node)
         {
+            var thisSide = node.Ancestors()
+                .TakeWhile(x => x.IsAnyKind(
+                    SyntaxKind.Argument,
+                    SyntaxKindEx.TupleExpression,
+                    SyntaxKindEx.SingleVariableDesignation,
+                    SyntaxKindEx.ParenthesizedVariableDesignation,
+                    SyntaxKindEx.DiscardDesignation,
+                    SyntaxKindEx.DeclarationExpression))
+                .LastOrDefault();
+            if ((TupleExpressionSyntaxWrapper.IsInstance(thisSide) || DeclarationExpressionSyntaxWrapper.IsInstance(thisSide))
+                && thisSide.Parent is AssignmentExpressionSyntax assignment)
+            {
+                var otherSide = assignment switch
+                {
+                    { Left: { } left, Right: { } right } when left.Equals(thisSide) => right,
```

Comment:
from a readability and maintainability point of view, it would be easier to understand two methods that do this:
- one that finds, given something on the left, its complement on the right
- one that finds, given something on the right, its complement on the left

Is there a reason why not to split this method into two? This would also make the tests easier to follow.

And the static local functions (Which are quite big ) could be private shared methods.

Reply:
The idea is to extend the method later for patterns, like so:
```cs
_ = (1, 2) is (var a, var b);  // mappings between a<->1 and b<->2
_ = (1, 2) is var (a, b);       // mappings between a<->1 and b<->2
_ = (1, 2) switch               // mapping between 1 -> [a1, a2] and a1 -> 1 and 2 <-> []
{
  (var a1, _) when a > 10 => 10,
  (var a2, _) => a2,
};
```
We would end up with half a dozen methods that do all the same if we make the distinction between left and right, or expression and pattern or designation and element, and so on.
The signature has to change for switch expressions to return a list of nodes, but the name and the general idea could stay the same.